mathics/builtin/numbers/randomnumbers.py

```diff
@@ -497,11 +499,21 @@ class RandomComplex(Builtin):
 
     @staticmethod
     def to_complex(value, evaluation):
-        result = value.to_python(n_evaluation=evaluation)
-        if isinstance(result, (float, int)):
-            result = complex(result)
-        if isinstance(result, complex):
-            return result
+        result = eval_N(value, evaluation)
+
+        if hasattr(result, "value"):
+            result_value = result.value
+        else:
+            # TODO: result.value does not work, because
+            # Complex does not have a ``value`` attribute.
```

Comment:
I have been thinking about this. When the complex value is known to be exactly the same, for example we have an integer imaginary number that is not too large, or the "float" representations of the real and imaginary parts fit exactly in the float number, then Complex could have a Python value which could be Python datatype complex or integer (and possibly float if the float representation is exact). 

However this kind of thing, especially at this stage of development is to involved seems like a rather custom kind of things and probably not the main source of slowdowns, that I don't we should undertake this activity just yet.

Reply:
Indeed, this is why I just left a comment...

Reply:
It is not just Complex though. And it is not clear that adding value to complex would handle everything either. Python3 has Factions as well, and fixed-point numbers. And even with this there may be other Atoms that don't have values. 

So from my point of view this is a little vague. It is not that Complex _alone_ doesn't have a value so much as right now value is optional. And that is what the test says.

Reply:
The comment was about `Complex` in a very particular context. If this class had a `value` property (that could even be calculated on the fly) then the `if` and the call to `to_python` would not be necessary.